src/app.js

```diff
@@ -1,12 +1,14 @@
 const express = require("express");
 const path = require("path");
+const session = require("express-session");
 const { configureNunjucks } = require("./config/nunjucks");
 const validateForm = require("./validateForm");
 const {
   setGa4ContainerId,
   setStatusCode,
   setTaxonomyValues,
   setPageTitle,
+  checkSessionAndRedirect,
```

Comment:
It's not a GTM middleware function. You have to put it in another file.

Reply:
done